### PR TITLE
far2l: unstable-2017-07-13 -> unstable-2017-09-30, fix darwin build

### PR DESCRIPTION
**Pull request by volth**
- [x] minor bugfixes ([changelog](https://github.com/elfmz/far2l/commits/master))
- [x] support build on darwin

### CODE REVIEW

**Comment by volth** [comment]
@LnL7

File: pkgs/applications/misc/far2l/default.nix [hunk 1]
```diff
@@ -1,29 +1,34 @@
...
-  postPatch = ''
-    echo 'echo ${build}' > far2l/bootstrap/scripts/vbuild.sh
-
-    substituteInPlace far2l/bootstrap/open.sh              \
+  postPatch = optionalString stdenv.isLinux ''
+    substituteInPlace far2l/bootstrap/open.sh \
       --replace 'gvfs-trash'  '${gvfs}/bin/gvfs-trash'
+  '' + optionalString stdenv.isDarwin ''
+    substituteInPlace far2l/CMakeLists.txt \
+      --replace "-framework System" -lSystem
```
Comment (LnL7) [inline pkgs/applications/misc/far2l/default.nix]: Is this still necessary if you include `darwin.apple_sdk.frameworks.System`?

Comment (volth) [reply pkgs/applications/misc/far2l/default.nix]: No idea. I have absolutely no dev and almost no user experience on Darwin, just fixed "library not found" [after](https://github.com/NixOS/nixpkgs/blob/7ebcd39a0f147ffa544818c9fd5954bc9dcae2d8/pkgs/development/libraries/wxwidgets/3.0/default.nix#L69) [wxWidgets](https://github.com/NixOS/nixpkgs/blob/7ebcd39a0f147ffa544818c9fd5954bc9dcae2d8/pkgs/development/libraries/wxwidgets/3.0/mac.nix#L63) which ```far2l``` is using

Comment (volth) [reply pkgs/applications/misc/far2l/default.nix]: And... there is no ```System``` in ```darwin.apple_sdk.frameworks```: https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/darwin/apple-sdk/frameworks.nix

Comment (LnL7) [reply pkgs/applications/misc/far2l/default.nix]: oh, I thought I added that recently

File: pkgs/applications/misc/far2l/default.nix [hunk 2]
```diff
@@ -1,29 +1,34 @@
...
 
   patches = [ ./add-nix-syntax-highlighting.patch ];
 
-  postPatch = ''
-    echo 'echo ${build}' > far2l/bootstrap/scripts/vbuild.sh
-
-    substituteInPlace far2l/bootstrap/open.sh              \
+  postPatch = optionalString stdenv.isLinux ''
+    substituteInPlace far2l/bootstrap/open.sh \
       --replace 'gvfs-trash'  '${gvfs}/bin/gvfs-trash'
```
Comment (LnL7) [inline pkgs/applications/misc/far2l/default.nix]: gfvs doesn't work on darwin because it depends on systemd

**Pull request merged**